### PR TITLE
Check for oneclick access in manual review reminders

### PR DESCRIPTION
**Pull request by ajnyga**
Explained here: https://github.com/pkp/pkp-lib/issues/2410

@asmecher, the `initdata` part maybe needs some revising, but this is working already as is.

### CODE REVIEW

File: controllers/grid/users/reviewer/form/ReviewReminderForm.inc.php [hunk 1]
```diff
@@ -141,15 +144,26 @@ function execute($args, $request) {
...
-		$email = new SubmissionMailTemplate($submission, 'REVIEW_REMIND', null, null, null, false);
+		$context = $request->getContext();
+		$templateKey = $this->_getMailTemplateKey($context);	
+		$email = new SubmissionMailTemplate($submission, $templateKey, null, null, null, false);
+		
+		$reviewUrlArgs = array('submissionId' => $reviewAssignment->getSubmissionId());
+		if ($context->getSetting('reviewerAccessKeysEnabled')) {
+			import('lib.pkp.classes.security.AccessKeyManager');
+			$accessKeyManager = new AccessKeyManager();
+			$expiryDays = $context->getSetting('numWeeksPerReview') + 4 * 7;
```
Comment (asmecher) [inline controllers/grid/users/reviewer/form/ReviewReminderForm.inc.php]: @ajnyga, would you mind adding the missing parentheses here too? Should be:

    $expiryDays = ($context->getSetting('numWeeksPerReview') + 4) * 7;

File: controllers/grid/users/reviewer/form/ReviewReminderForm.inc.php [hunk 2]
```diff
@@ -160,6 +174,23 @@ function execute($args, $request) {
 		$reviewAssignmentDao = DAORegistry::getDAO('ReviewAssignmentDAO');
 		$reviewAssignmentDao->updateObject($reviewAssignment);
 	}
+
+	/**
+	 * Get the email template key depending on if reviewer one click access is
+	 * enabled or not.
+	 *
+	 * @param mixed $context Context
```
Comment (asmecher) [inline controllers/grid/users/reviewer/form/ReviewReminderForm.inc.php]: We use Doxygen for automatic code documentation, and use...

```
@param $paramName TypeName An English description of the parameter
```

**Comment by ajnyga** [comment]
@asmecher did the math and fixed the param. I am actually not sure if I fixed it the way you were thinking, but if I did, then there is a similar issue here https://github.com/pkp/pkp-lib/blob/master/controllers/grid/users/reviewer/form/ReviewerForm.inc.php#L454

**Comment by asmecher** [comment]
@ajnyga, can you double-check that line number?

**Comment by ajnyga** [comment]
I mean in relation to your earlier comment "We use Doxygen for automatic code documentation, and use... @param $paramName TypeName An English description of the parameter"

If the "@param mixed  $context Context" part was the problem you were thinking there? Or alternatively I totally misunderstood how to write the param description correctly. I mean, should that be "@param $context Context"?

**Comment by asmecher** [comment]
Ah, sorry, I misread your comment. It's almost there --

     @param mixed $context Context

...should be perhaps...

     @param $context Context The user's current context (journal/press).

(`Context`, rather than `mixed`, is the type description.)

**Comment by ajnyga** [comment]
ok, I think that "@param mixed $context Context" is at least in few similar places.

I already closed my other computer so I will get back to this tomorrow. Or if you like you can of course edit the param and merge this.

**Comment by asmecher** [comment]
Thanks, @ajnyga -- yes, it's likely that some of the syntax here isn't correct. I'll merge this and take care of it.

**Comment by ajnyga** [comment]
I noticed that you did not merge this yet, so I made the requested changes. Also fixed the othe "param mixed" I found. Left out the (journal/press) to make room for forthcoming OCS 3.x and other PKP products you do not know about yet.

**Comment by asmecher** [comment]
Merged, thanks!

> other PKP products you do not know about yet.

... I see you have plans for us... Concretely, we are planning on rewriting OCS, and who knows what else.

**Pull request merged**